soc/riscv/riscv-privilege/efinix-sapphire/Kconfig.soc

```diff
@@ -0,0 +1,18 @@
+# Copyright (c) 2023 Efinix Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+choice
+	prompt "Efinix SoC selection"
+	depends on SOC_SERIES_EFINIX_SAPPHIRE
+
+config SOC_RISCV32_EFINIX_SAPPHIRE
+	bool "Efinix Sapphire VexRiscv system implementation"
+	select ATOMIC_OPERATIONS_C
```

Comment:
Are you sure this should not be `ATOMIC_OPERATIONS_BUILTIN` instead?

Reply:
Thank you for your suggestion. However, several RISC-V privilege SoCs, including the ones listed below, currently utilize ATOMIC_OPERATIONS_C: (I'm **not sure** why!)

```
soc/riscv/litex-vexriscv/Kconfig.soc:   select ATOMIC_OPERATIONS_C
soc/riscv/openisa_rv32m1/Kconfig.soc:   select ATOMIC_OPERATIONS_C
soc/riscv/riscv-privilege/opentitan/Kconfig.soc:        select ATOMIC_OPERATIONS_C
soc/riscv/riscv-privilege/neorv32/Kconfig.soc:  select ATOMIC_OPERATIONS_C
soc/riscv/riscv-privilege/sifive-freedom/Kconfig.soc:   select ATOMIC_OPERATIONS_C
soc/riscv/riscv-privilege/sifive-freedom/Kconfig.soc:   select ATOMIC_OPERATIONS_C
soc/riscv/riscv-privilege/sifive-freedom/Kconfig.soc:   select ATOMIC_OPERATIONS_C
soc/riscv/riscv-privilege/niosv/Kconfig.soc:    select ATOMIC_OPERATIONS_C
soc/riscv/riscv-privilege/gd32vf103/Kconfig.series:     select ATOMIC_OPERATIONS_C
soc/riscv/esp32c3/Kconfig.defconfig:config ATOMIC_OPERATIONS_C
```

I've attempted to implement ATOMIC_OPERATIONS_BUILTIN for the current SoC in the PR but encountered an issue: during startup, it is causing a exception to `_Fault()`

Therefore, for the time being, we might need to continue with `ATOMIC_OPERATIONS_C` unless we have a good answer.

Reply:
@npitre given #57674 do you have any idea about this?

Reply:
This is strange I only see the issue when I enable `CONFIG_SHELL=y` along with `ATOMIC_OPERATIONS_BUILTIN`

Reply:
@carlocaione This is done now, turns out I made an config mistake on the soft SoC. 

Things are working fine now.